numpy/_core/src/multiarray/stringdtype/dtype.c

```diff
@@ -709,13 +709,13 @@ stringdtype_repr(PyArray_StringDTypeObject *self)
 }
 
 static PyObject *
-stringdtype__reduce__(PyArray_StringDTypeObject *self)
+stringdtype__reduce__(PyArray_StringDTypeObject *self, PyObject *NPY_UNUSED(args))
 {
     PyObject *ret = NULL;
     PyObject *obj = NULL;
     PyObject *mod = NULL;
 
-    ret = PyTuple_New(3);
+    ret = PyTuple_New(2);
```

Comment:
Two random comments not important:
1. You can use `npy_cache_import("numpy._core._internal", "_convert_to_stringdtype_kwargs", &_convert_to_stringdtype_kwargs)` (with `static PyObject *_convrt_to_stringdtype_kwargs = NULL;`)
2. Could consider using `Py_BuildValue("O(iO)", _convert_to_stringdtype_kwargs, self->coerce, self->na_object);` if just not to worry about the tuple refercounting?